cmd/parser/parse.go

```diff
@@ -0,0 +1,167 @@
+	cobraCmd.Flags().StringVarP(&cmd.ruleFilePath, "file", "f", "", "Read rules from a file pointed to by parameter value. The file must contain a valid yaml list, where each rule entry starts with '-' and is placed in its own line.")
+	cobraCmd.Flags().BoolVarP(&cmd.sort, "priority", "p", false, "Sort rule entries by their priority, in descending priority order.")
+	cobraCmd.Flags().BoolVarP(&cmd.unique, "unique", "u", false, "Display only non duplicated rules. Error entries are not considered for uniqueness.")
+	cobraCmd.Flags().BoolVarP(&cmd.signature, "signature", "s", false, "Mark rules with the mirrored signatures as duplicated. For example aws(PR=*, HR=westeurope) and aws(PR=westeurope, HR=*) are considered duplicated because of having mirrored signatures.")
+	cobraCmd.Flags().BoolVarP(&cmd.noColor, "no-color", "n", false, "Disable use color characters when generating output.")
+	cobraCmd.MarkFlagsOneRequired("entry", "file")
+
+	return cobraCmd
+}
+
+type ProcessingPair struct {
+	ParsingResults  *rules.ParsingResult
+	MatchingResults *rules.MatchingResult
+}
+
+func (cmd *ParseCommand) Run() error {
+
+	printer := rules.NewColored(cmd.cobraCmd.Printf)
+	if cmd.noColor {
+		printer = rules.NewNoColor(cmd.cobraCmd.Printf)
+	}
+
+	if cmd.match != "" && (!cmd.sort || !cmd.unique) {
+		cmd.cobraCmd.Printf("\tMatching is only supported when both priority and uniqueness flags are specified.\n")
+		return nil
+	}
+
+	var rulesService *rules.RulesService
+	var err error
+	if cmd.ruleFilePath != "" {
+		cmd.cobraCmd.Printf("Parsing rules from file: %s\n", cmd.ruleFilePath)
+		rulesService, err = rules.NewRulesServiceFromFile(cmd.ruleFilePath, cmd.sort, cmd.unique, cmd.signature)
+	} else {
+		rulesService, err = rules.NewRulesServiceFromString(cmd.rule, cmd.sort, cmd.unique, cmd.signature)
+	}
+
+	if err != nil {
+		cmd.cobraCmd.Printf("Error: %s\n", err)
+		return nil
+	}
+
+	var testDataForMatching *rules.MatchableAttributes
+	if cmd.match != "" {
+		testDataForMatching = getTestData(cmd.match)
+	}
+
+	var matchingResults map[uuid.UUID]*rules.MatchingResult
+	if cmd.match != "" && testDataForMatching != nil {
+		matchingResults = rulesService.Match(testDataForMatching)
+	}
+
+	printer.Print(rulesService.Parsed.Results, matchingResults)
+
+	hasErrors := false
+	for _, result := range rulesService.Parsed.Results {
+		if result.HasErrors() {
+			hasErrors = true
+			break
+		}
+	}
+
+	if hasErrors {
+		cmd.cobraCmd.Printf("There are errors in your rule configuration. Fix above errors in your rule configuration and try again.\n")
+		return nil
+	}
+
+	return nil
+}
+
+type conf struct {
+	Rules []string `yaml:"rule"`
+}
+
+func (c *conf) getConf(file string) *conf {
+	yamlFile, err := os.ReadFile(file)
+	if err != nil {
+		log.Printf("yamlFile.Get err   #%v ", err)
+	}
+	err = yaml.Unmarshal(yamlFile, c)
+	if err != nil {
+		log.Fatalf("Unmarshal: %v", err)
+	}
+
+	return c
+}
+
+func getTestData(content string) *rules.MatchableAttributes {
```

Comment:
Shouldn't we place it in test code?

Reply:
https://github.com/kyma-project/kyma-environment-broker/pull/1811/commits/ca372d14eb7e029f27b638167a79911598227667